client/apiversion_test.go

```diff
@@ -0,0 +1,164 @@
+package client_test
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	gc "gopkg.in/check.v1"
+)
+
+type versionHandler struct {
+	authBody string
+	port     string
+}
+
+type makeServiceURLTest struct {
+	serviceType string
+	version     string
+	parts       []string
+	success     bool
+	URL         string
+	err         string
+}
+
+func (s *localLiveSuite) makeServiceURLTests() []makeServiceURLTest {
+	return []makeServiceURLTest{
+		{
+			serviceType: "compute",
+			version:     "v2.1",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.1/foo/bar/",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2.0",
+			parts:       []string{},
+			success:     true,
+			URL:         "http://localhost:%s/v2.0",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2.0",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.0/foo/bar/",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.1/foo/bar/",
+		},
+		{
+			serviceType: "object-store",
+			version:     "",
+			parts:       []string{"foo", "bar"},
+			success:     true,
+			URL:         "http://localhost:%s/swift/v1/foo/bar",
+		},
+		{
+			serviceType: "object-store",
+			version:     "v2.0",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/swift/v1/foo/bar/",
```

Comment:
The endpoint says v1, however the API doesn't have a call that returns a list of versions. Here's a link to the API: http://developer.openstack.org/api-ref/object-storage/

For object-storage in MakeServiceURL - the version is ignored.  As far as I'm aware there is not version 2 of the object-store (Swift) API.  It could be that they will create a new endpoint service type if one is needed.  There is already volume and volumev2 service types for Cinder.

You do need to be aware of OpenStack services and version when making calling SendRequest and/or MakeServiceURL.